src/classes/BDI_DataImportBatch_SEL.cls

```diff
@@ -51,8 +52,46 @@ public with sharing class BDI_DataImportBatch_SEL {
                     Donation_Matching_Behavior__c,
                     Donation_Matching_Implementing_Class__c,
                     Donation_Matching_Rule__c,
-                    Donation_Date_Range__c
+                    Donation_Date_Range__c,
+                    Process_with_Scheduled_Job__c
             FROM DataImportBatch__c
             WHERE Id in :setBatchId];
     }
+
+    /*******************************************************************************************************
+    * @description queries the database for the Data Import Batch record with the specified Id
+    * @param batchId The batch to query for.
+    * @return DataImportBatch__c
+    */
+    public static DataImportBatch__c selectById(Id batchId) {
+        list<DataImportBatch__c> listBatches = selectByIds(new set<Id>{batchId});
+        return listBatches[0];
+    }
+
+    /*******************************************************************************************************
+    * @description queries the database for all Data Import Batch records that are marked with Scheduled
+    * processing, and have DI records to import.
+    * @return list<Id>
+    */
+    public static list<Id> selectWithScheduledProcessing() {
+        List<Id> listIdBatches = new List<Id>();
+
+        for (DataImportBatch__c diBatch : [SELECT
+            Id,
```

Comment:
should we go with field name "Process with Scheduled Job" with the values "To be processed" and "Completed"?  @smarovich any better wording you suggest for both the field name and values?